src/MICore/LaunchOptions.xsd

```diff
@@ -399,6 +389,23 @@
         </xs:documentation>
       </xs:annotation>
     </xs:attribute>
+    <xs:attribute name="ProcessId" type="xs:int" use="optional">
+      <xs:annotation>
+        <xs:documentation>If supplied, the debugger will attach to the process rather than launching a new one. Note that some operating systems will require admin rights to do this.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="CoreDumpPath" type="xs:string" use="optional">
+      <xs:annotation>
+        <xs:documentation>Path to a core dump file for the specified executable.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="DebugChildProcesses" type="xs:boolean" use="optional">
+      <xs:annotation>
+        <xs:documentation>
+          If true, automatically attach debugger to any child processes created.
+        </xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
```

Comment:
I don’t see any reference to  BaseLaunchOptions from either IOSLaunchOptions or AndroidLaunchOptions. How do these end up applying?

From: Gregg Miskelly [mailto:notifications@github.com]
Sent: Friday, April 8, 2016 6:13 PM
To: Microsoft/MIEngine <MIEngine@noreply.github.com>
Cc: Paul Maybee <paulmay@microsoft.com>
Subject: Re: [Microsoft/MIEngine] When the underlying debugger supports it, automatically attach to chi… (#259)


In src/MICore/LaunchOptions.xsd<https://github.com/Microsoft/MIEngine/pull/259#discussion_r59107175>:

> +      <xs:annotation>

> +        <xs:documentation>If supplied, the debugger will attach to the process rather than launching a new one. Note that some operating systems will require admin rights to do this.</xs:documentation>

> +      </xs:annotation>

> +    </xs:attribute>

> +    <xs:attribute name="CoreDumpPath" type="xs:string" use="optional">

> +      <xs:annotation>

> +        <xs:documentation>Path to a core dump file for the specified executable.</xs:documentation>

> +      </xs:annotation>

> +    </xs:attribute>

> +    <xs:attribute name="DebugChildProcesses" type="xs:boolean" use="optional">

> +      <xs:annotation>

> +        <xs:documentation>

> +          If true, automatically attach debugger to any child processes created.

> +        </xs:documentation>

> +      </xs:annotation>

> +    </xs:attribute>

This is going to wind up applying to Android and iOS also. Should this really be just pipe/tcp/local?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/MIEngine/pull/259/files/e954491c978a954faaadea93787112899af6f1c3#r59107175>